### ISSUE TITLE
Toggle dark mode in android navigation bar from TWA

### ISSUE THREAD

**Issue by ashiqdey** [GoogleChrome/android-browser-helper#382]
TWA is awesome but my TWA supports light and dark mode, so I want to communicate from TWA to android to toggle the navigation bar's color

**Describe the solution you'd like**
Any communication bridge between the two without page reload would be much appriciated

**Describe alternatives you've considered**
Or in browser we can `color-scheme: dark;` to toggle the color of scroll and other native items, something same would also work. 

**Additional context**
Its too irritating to have ta app in dark color and the navigation bar is in white

**Comment by PEConn** [comment]
You can set the colour of the navigation bar while the phone is in dark mode using [android.support.customtabs.trusted.NAVIGATION_BAR_COLOR_DARK](https://github.com/GoogleChrome/android-browser-helper/blob/main/androidbrowserhelper/src/main/java/com/google/androidbrowserhelper/trusted/LauncherActivityMetadata.java#L69) in the manifest, [like this](https://github.com/GoogleChrome/android-browser-helper/blob/6be9bb7e3fc5bf0bd228ea4e3ad6e5649e9c218f/demos/twa-basic/src/main/AndroidManifest.xml#L47).

Is that what you want to do?

**Comment by ashiqdey** [comment]
> You can set the colour of the navigation bar while the phone is in dark mode using [android.support.customtabs.trusted.NAVIGATION_BAR_COLOR_DARK](https://github.com/GoogleChrome/android-browser-helper/blob/main/androidbrowserhelper/src/main/java/com/google/androidbrowserhelper/trusted/LauncherActivityMetadata.java#L69) in the manifest, [like this](https://github.com/GoogleChrome/android-browser-helper/blob/6be9bb7e3fc5bf0bd228ea4e3ad6e5649e9c218f/demos/twa-basic/src/main/AndroidManifest.xml#L47).
> 
> Is that what you want to do?

No, I want to able to toggle the navigation color based on my TWA, if my TWA is in dark mode I want to toggle nav. bar to dark and vice versa. I need some kind of bridge between the TWA and android.

**Comment by PEConn** [comment]
Oh right.

So for the Android status bar (the one at the top), Chrome lets that be set
to the website's theme-color. We could do the same for navigation bars, but
I'm not sure if that's the right decision here.

We'll look into this further, but it seems unlikely we'll add this as a
feature soon, so if this is a pressing issue, I'd recommend working around
it by using a darker navigation bar colour in both light and dark mode.

**Comment by limhenry** [comment]
I am having the same issue too. In this case, is `postMessage` would be useful [if the feature is enabled](https://github.com/GoogleChrome/android-browser-helper/issues/55)?

Btw, is it possible to listen to the URL changes from Android/Java side?
